epochStart/bootstrap/syncValidatorStatus_test.go

```diff
@@ -95,6 +99,148 @@ func TestSyncValidatorStatus_NodesConfigFromMetaBlock(t *testing.T) {
 	require.NotNil(t, registry)
 }
 
+func TestSyncValidatorStatus_processValidatorChangesFor(t *testing.T) {
+	t.Parallel()
+
+	args := getSyncValidatorStatusArgs()
+
+	mbHeaderHash1 := []byte("mb-hash1")
+	mbHeaderHash2 := []byte("mb-hash2")
+
+	metaBlock := &block.MetaBlock{
+		Nonce: 10,
+		Epoch: 1,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{
+				Hash: mbHeaderHash1,
+				Type: block.TxBlock,
+			},
+			{
+				Hash: mbHeaderHash2,
+				Type: block.PeerBlock,
+			},
+		},
+	}
+
+	expectedBody := &block.Body{
+		MiniBlocks: []*block.MiniBlock{
+			{
+				ReceiverShardID: 1,
+				SenderShardID:   0,
+			},
+		},
+	}
+
+	svs, _ := NewSyncValidatorStatus(args)
+
+	wasCalled := false
+	svs.nodeCoordinator = &mock.NodesCoordinatorStub{
+		EpochStartPrepareCalled: func(metaHdr data.HeaderHandler, body data.BodyHandler) {
+			wasCalled = true
+			assert.Equal(t, metaBlock, metaHdr)
+			assert.Equal(t, expectedBody, body)
+		},
+	}
+	svs.miniBlocksSyncer = &epochStartMocks.PendingMiniBlockSyncHandlerStub{
+		SyncPendingMiniBlocksCalled: func(miniBlockHeaders []data.MiniBlockHeaderHandler, ctx context.Context) error {
+			return nil
+		},
+		GetMiniBlocksCalled: func() (map[string]*block.MiniBlock, error) {
+			return map[string]*block.MiniBlock{
+				string(mbHeaderHash2): {
+					ReceiverShardID: 1,
+					SenderShardID:   0,
+				},
+			}, nil
+		},
+	}
+
+	err := svs.processValidatorChangesFor(metaBlock)
+	require.NoError(t, err)
+	assert.True(t, wasCalled)
+}
+
+func TestSyncValidatorStatus_findPeerMiniBlockHeaders(t *testing.T) {
+	t.Parallel()
+
+	metaBlock := &block.MetaBlock{
+		Nonce: 37,
+		Epoch: 0,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{
+				Hash: []byte("mb-hash1"),
+				Type: block.TxBlock,
+			},
+			{
+				Hash: []byte("mb-hash2"),
```

Comment:
can you create the miniBlock first and reference it here and in the expectedMbHeaders?

Reply:
:+1: